dynamic_prompting.py

```diff
@@ -17,13 +17,97 @@
 
 WILDCARD_DIR = getattr(opts, "wildcard_dir", "scripts/wildcards")
 MAX_RECURSIONS = 20
-VERSION = "0.5.0"
+VERSION = "0.6.0"
+WILDCARD_SUFFIX = "txt"
 
 re_wildcard = re.compile(r"__(.*?)__")
 re_combinations = re.compile(r"\{([^{}]*)}")
 
 DEFAULT_NUM_COMBINATIONS = 1
 
+class WildcardFile:
+    def __init__(self, path: Path, encoding="utf8"):
+        self._path = path
+        self._encoding = encoding
+
+    def get_wildcards(self) -> Set[str]:
+        is_empty_line = lambda line: line is None or line.strip() == "" or line.strip().startswith("#")
+
+        with self._path.open(encoding=self._encoding, errors="ignore") as f:
+            lines = [line.strip() for line in f if not is_empty_line(line)]
+            return set(lines)
+
+
+class WildcardManager:
+    def __init__(self, path:str=WILDCARD_DIR):
+        self._path = Path(path)
+
+    def _directory_exists(self) -> bool:
+        return self._path.exists() and self._path.is_dir()
+
+    def ensure_directory(self) -> bool:
+        try:
+            self._path.mkdir(parents=True, exist_ok=True)
+        except Exception as e:
+            logger.exception(f"Failed to create directory {self._path}")
+
+    def get_files(self, relative:bool=False) -> list():
+        if not self._directory_exists():
+            return []
+
+        files = self._path.rglob(f"*.{WILDCARD_SUFFIX}")
+        if relative:
+            files = [f.relative_to(self._path) for f in files]
+
+        return files
+
+    def match_files(self, wildcard:str) -> list():
+        return [
+            WildcardFile(path) for path in self._path.rglob(f"{wildcard}.{WILDCARD_SUFFIX}")
+        ]
+
+    def get_wildcards(self) -> list():
+        files = self.get_files(relative=True)
+        wildcards = [f"__{path.with_suffix('')}__" for path in files]
+        return wildcards
+
+class UiCreation:
+    def write(self, path):
+        if path.is_dir():
+            return self.write_dir(path)
+        else:
+            return self.write_txt(path)        
+
+    def write_txt(self, path):
+        temp = ""
+        filename = path.name
+        wildcard = "__" + "/".join((str(path).split("\\"))[2:-1])+ "/" + filename.replace(".txt", "") + "__"
```

Comment:
WildcardManager.get_wildcards() might be better here, that way you can avoid manipulating the path yourself. For example, since you're on Windows, you are assuming that the path separator is a backslash '\', I run linux and my path separators are '/' - if at all possible, we should concentrate any path manipulation in WildcardManager. Let me know if there are any additional methods that you may need.

Reply:
By using `os.sep` instead of `\\` worked for me so it should work on all platforms.
Dont know any other way around the manipulation.